OPAL/tac/src/main/scala/org/opalj/tac/cg/android/AndroidICCAnalysis.scala

```diff
@@ -0,0 +1,1467 @@
+    final val setPackage = "setPackage"
+
+    final val onStart = "onStart"
+    final val onPause = "onPause"
+    final val onStop = "onStop"
+    final val onDestroy = "onDestroy"
+    final val onActivityResult = "onActivityResult"
+    final val onResume = "onResume"
+    final val onRestart = "onRestart"
+    final val onStartCommand = "onStartCommand"
+    final val startService = "startService"
+    final val onBind = "onBind"
+    final val onRebind = "onRebind"
+    final val onUnbind = "onUnbind"
+    final val bindService = "bindService"
+    final val stopService = "stopService"
+    final val unbindService = "unbindService"
+    final val stopSelf = "stopSelf"
+    final val onReceive = "onReceive"
+    final val onCreate = "onCreate"
+    final val onBroadcastReceive = "onBroadcastReceive"
+
+    //explicit
+    final val setClass = "setClass"
+    final val setClassName = "setClassName"
+    final val setComponent = "setComponent"
+
+    final val senderClasses = List(
+        contextOT, pendingIntentOT, ObjectType("android/content/IntentSender"), androidActivityOT
+    )
+    final val virtualSendingMethods = List(
+        "startIntentSender", "startIntentSenderForResult", "startIntentSenderFromChild", "startActivityFromChild",
+        "startActivity", "startActivities", "startActivityForResult", "startNextMatchingActivity", "sendIntent", "send",
+        "bindIsolatedService", bindService, "bindServiceAsUser", "sendBroadcast", "sendBroadcastAsUser",
+        "sendBroadcastWithMultiplePermissions", "sendOrderedBroadcast", "sendOrderedBroadcastAsUser",
+        "sendStickyBroadcast", "sendStickyBroadcastAsUser", "sendStickyOrderedBroadcast", "sendStickyOrderedBroadcastAsUser",
+        "startForegroundService", startService
+    )
+
+    final val staticSendingMethods = List(
+        "getActivities", "getActivity", "getBroadcast", "getForegroundService"
+    )
+
+    val activityStartMethods: List[String] = List(
+        "startActivity", "startActivityForResult", "startActivityFromChild", "startActivityIfNeeded", "startActivities",
+        "getActivity", "getActivities", "startNextMatchingActivity"
+    )
+    val serviceMethods: List[String] = List(
+        startService, bindService, stopService, unbindService, "bindIsolatedService",
+        "bindServiceAsUser", "startForegroundService"
+    )
+    val broadcastReceiverStartMethods: List[String] = List(
+        "sendBroadcast", "sendOrderedBroadcast", "sendStickyBroadcast", "sendStickyOrderedBroadcast", "getBroadcast",
+        "sendBroadcastAsUser", "sendStickyBroadcastAsUser", "sendStickyOrderedBroadcastAsUser"
+    )
+
+    final val intentMethods = List(
+        setAction, addCategory, setData, setDataAndNormalize, setDataAndType, setDataAndTypeAndNormalize, setType, setTypeAndNormalize, setPackage
+    )
+    final val explicitMethods = List(setClass, setClassName, setComponent)
+
+    val intentFilters: ListBuffer[IntentFilter] = ListBuffer.empty[IntentFilter]
```

Comment:
All of these mutable `ListBuffer` and `Map`s seem strange. Please consider refactoring to immutable data structures that are initialized a single time. Either in the constructor of this class, or if that is not possible, maybe encapsulate these data structures into an immutable state class as an inner class here. Avoid mutable data structures as far as possible.